OMCompiler/Compiler/FrontEnd/SCodeUtil.mo

```diff
@@ -5089,18 +5084,10 @@ functions, destructor and constructor. "
   input  list<SCode.Element> els;
   output Boolean res;
 algorithm
-  res := matchcontinue(els)
-    case _
-      equation
-        3 = listLength(els);
-        true = hasExtendsOfExternalObject(els);
-        true = hasExternalObjectDestructor(els);
-        true = hasExternalObjectConstructor(els);
-      then
-        true;
-
-    else false;
-  end matchcontinue;
+  res := hasExtendsOfExternalObject(els)
+         and hasExternalObjectDestructor(els)
+         and hasExternalObjectConstructor(els)
+         and listLength(els) == 3;
```

Comment:
Note that Modelica has no short-circuit evaluation. You will want to use if-expressions to speed this up (and start with listLength(els) == 3).